app/validation/validator.py

```diff
@@ -149,8 +155,12 @@ def validate_routing_on_answer_options(self, block, answer):
             for rule in block['routing_rules']:
                 if 'goto' in rule and 'when' in rule['goto'].keys():
                     when = rule['goto']['when']
-                    if 'id' in when and when['id'] == answer['id'] and when['value'] in options:
-                        options.remove(when['value'])
+                    if 'id' in when[0]:
+                        if not self.validate_when_rule(when, answer_ids):
+                            answer_errors.append(self._error_message('The answer id - {} in the "when" clause for {} '
+                                                                     'does not exist'.format(when[0]['id'], block['id'])))
+                        if when[0]['id'] == answer['id'] and when[0]['value'] in options:
+                            options.remove(when[0]['value'])
```

Comment:
I know this wasn't introduced in this change. But what is this bit of code doing? Why is it removing the value from the answer options if it's used in the routing rules. If you know, could you add a comment please?

Reply:
@nealedj For some reason it checks answers options, and takes away any values that match in when rule, the result then is used to evaluate if there are unrouted options in the answer see line `170`. It seems a bit hacky to be honest. Maybe tomorrow we could go over it and change appropriately?

Reply:
I think thats why you were seeing those errors, also i dont think this takes into account default routing, i.e. when it just routes if no other `when` is evaluated true